aten/src/ATen/core/op_registration/op_registration.cpp

```diff
@@ -15,7 +15,7 @@ class RegisterOperators::OperatorRegistrar final {
   explicit OperatorRegistrar(FunctionSchema&& schema, c10::optional<TensorTypeId> dispatch_key, KernelFunction* kernel, KernelCacheCreatorFunction&& cache_creator)
   : op_(Dispatcher::singleton().registerSchema(std::move(schema))), kernel_registration_handle_(c10::nullopt) {
     // either both, kernel and cache_creator, or none must be set.
-    AT_ASSERT((kernel != nullptr) == static_cast<bool>(cache_creator));
+    TORCH_INTERNAL_ASSERT((kernel != nullptr) == static_cast<bool>(cache_creator));
```

Comment:
This is in c10 namespace, so I guess it is public API. Isn't this assert invalid?

Reply:
This class is defined in a .cpp file, not in a header. Anything outside this .cpp file can't call it.